.github/workflows/ci.yml

```diff
@@ -146,69 +146,100 @@ jobs:
-    - uses: actions/cache@v3
-      continue-on-error: true
-      timeout-minutes: 2
-      with:
-        path: ~/go/pkg/mod
-        key: ${{ runner.os }}-go-pkg-mod-${{ hashFiles('go.sum') }}      
-        
-    - name: Get dependencies
-      run: |
-        go mod download
-
-    - name: Acceptance tests
-      timeout-minutes: 10
-      env:
-        TWINGATE_URL: ${{ secrets.TWINGATE_URL }}
-        TWINGATE_NETWORK: ${{ secrets.TWINGATE_NETWORK }}
-        TWINGATE_API_TOKEN: ${{ secrets.TWINGATE_API_TOKEN }}
-        TEST_UNIQUE_VALUE: ${{ github.run_id }}-${{ github.run_number }}-${{ matrix.terraform }}
-      run: |
-        make testacc
-
-    - name: Send coverage
-      uses: shogo82148/actions-goveralls@v1
-      with:
-        path-to-profile: "./test/out/coverage.out"
-        flag-name: tests-acc-${{ matrix.terraform }}
-        parallel: true
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+        id: go
+
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ matrix.terraform }}
+          terraform_wrapper: false
+
+      - name: go env
+        run: echo "GOCACHE=$(go env GOCACHE)" >> $GITHUB_ENV
+      - uses: actions/cache@v3
+        continue-on-error: true
+        timeout-minutes: 2
+        with:
+          path: ${{ env.GOCACHE }}
+          key: ${{ runner.os }}-GOCACHE-${{ hashFiles('go.sum') }}-${{ hashFiles('internal/**') }}
+      - uses: actions/cache@v3
+        continue-on-error: true
+        timeout-minutes: 2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-pkg-mod-${{ hashFiles('go.sum') }}
+
+      - name: Get dependencies
+        run: |
+          go mod download
+
+      - name: Acceptance tests
+        timeout-minutes: 10
+        env:
+          TWINGATE_URL: ${{ secrets.TWINGATE_URL }}
+          TWINGATE_NETWORK: ${{ secrets.TWINGATE_NETWORK }}
+          TWINGATE_API_TOKEN: ${{ secrets.TWINGATE_API_TOKEN }}
+          TEST_UNIQUE_VALUE: ${{ github.run_id }}-${{ github.run_number }}-${{ matrix.terraform }}
+        run: |
+          make testacc
+
+      - name: Send coverage
+        uses: shogo82148/actions-goveralls@v1
+        with:
+          path-to-profile: "./test/out/coverage.out"
+          flag-name: tests-acc-${{ matrix.terraform }}
+          parallel: true
+
+  cleanup:
+    name: Cleanup
+    if: "!github.event.pull_request.head.repo.fork"
+    needs: tests-acceptance
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: 'go.mod'
+          cache: true
+        id: go
+
+      - name: Get dependencies
```

Comment:
Need to add the caching statements (see lines 172-185) here too

Reply:
thank you, added to smoke tests as well